src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_utils.py

```diff
@@ -79,12 +80,34 @@ def get_resource_id_by_storage_account_name(cli_ctx, storage_account_name):
     return arm_resource.id
 
 
-def get_resource_id_by_registry_name(cli_ctx, registry_name):
-    """Returns the resource id for the registry.
+def get_resource_id_by_registry_name(cli_ctx, registry_name, msg):
+    """Returns the resource id for the registry. If the registry cannot be found in the current subscription
+    then we prompt user to input resource ID for the registry.
+    Otherwise we return the resource ID returned from the ARM.
     :param str registry_name: The name of the registry
+    :param str msg: Prompt message for user to input the resourse ID
     """
-    arm_resource = _arm_get_resource_by_name(cli_ctx, registry_name, REGISTRY_RESOURCE_TYPE)
-    return arm_resource.id
+    result = get_resources_in_subscription(cli_ctx, REGISTRY_RESOURCE_TYPE)
+    elements = [item for item in result if item.name.lower() == registry_name.lower()]
+
+    if not elements or len(elements) != 1:
+        try:
+            resource_id = prompt(msg)
```

Comment:
This method would be confusing here, at least it shouldn't be in `_utils` as it won't be used by any other commands due to this special handling.

I would suggest that you do a `get_resource_id_by_registry_name` but instead of throwing, returning an empty result so you know the registry doesn't exist in the subscription.